### PR TITLE
New features for Ship based on Hull.io use case

### PR DESCRIPTION
**Pull request by unity**
– Better defaults
– Delays in Seconds instead of milliseconds (for Humans)
– Improved  titles and added description to all meaningful fields.
– Add "Reset Styles" toggle that allows to skip adding the `SHIP_ID *` directive, for easier overriding in client site
– Add "Display Login" toggle allowing to remove buttons to log in, making it a "Signup only" element
– Add "Display Signup" toggle allowing to remove buttons to sign up, making it a "Login only" element
– Add "Display User Profile" button allowing to hide ability to display profile screen
– Add "Custom Buttons" table allowing to display custom buttons on the page when user is logged in (Useful to display a link to a dashboard for instance ;p)
– Added some BEM css classes to allow css targeting
– Made Opening / Closing animation a bit lighter
– Fixed a few CSS issues (overflow for higher-than-screen forms, and borders for fieldset element)

### CODE REVIEW

**Comment by unity** [comment]
@citelao @oliger is the ship used in production by customers ?
If not it would be a good thing to change settings format.
Best practice is to group the settings sections so they look better in the dashboard

**Comment by oliger** [comment]
> Best practice is to group the settings sections so they look better in the dashboard

I agree but the way we group field at the moment does not seem right to me. The way we want to display them in the dashboard should not affect the way we store them. I prefer storing them flat for now.

File: src/lib/engine.js [hunk 1]
```diff
@@ -407,7 +407,7 @@ assign(Engine.prototype, EventEmitter.prototype, {
       this._activeSection = name;
 
       this.emitChange();
-    }, time);
+    }, time*1000);
```
Comment (oliger) [inline src/lib/engine.js]: space.

**Comment by oliger** [comment]
> Delays in Seconds instead of milliseconds (for Humans)

can we set float for number in the dashboard?

**Comment by citelao** [comment]
@oliger Shouldn't the manifest.json control how they are displayed? What is the alternative for storage/display control?

**Comment by oliger** [comment]
@citelao I think it should! but it describe how it is stored too.

I do not have alternative for now... But I know that having nested settings is error prone and by storing `properties` as a hash we cannot ensure the order of those properties.

**Comment by oliger** [comment]
@unity Let's keep your version. But it look strange to store two schemas in an array!

**Comment by citelao** [comment]
@unity @oliger settings and display order schema?

File: src/lib/engine.js (hunk 1 above)
Comment (unity) [inline src/lib/engine.js]: ??

**Comment by unity** [comment]
Reverted using groups in the manifest.

File: src/lib/engine.js (hunk 1 above)
Comment (oliger) [inline src/lib/engine.js]: Add space between `time`, `*` and `1000` :)

File: npm-debug.log [hunk 2]
```diff
@@ -0,0 +1,180 @@
+0 info it worked if it ends with ok
```
Comment (citelao) [inline npm-debug.log]: @unity can you .gitignore this?

Comment (unity) [inline npm-debug.log]: Indeed.

File: webpack.config.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ module.exports = {
   development:{
    browser: {
       name     : 'browser',
-      devtool  : '#inline-source-map',
+      // devtool  : '#inline-source-map',
```
Comment (citelao) [inline webpack.config.js]: This helps with debugging :sob:

File: npm-debug.log (hunk 2 above)
Comment (citelao) [inline npm-debug.log]: You might need to `git rm`

File: webpack.config.js (hunk 3 above)
Comment (citelao) [inline webpack.config.js]: Maybe `cheap-module-eval-source-map	` for development

http://webpack.github.io/docs/configuration.html#devtool

File: src/components/overlay/index.jsx [hunk 4]
```diff
@@ -231,12 +216,17 @@ export default React.createClass({
     };
   },
 
+  getClassName(){
+    return cx({[this.props.className]:true, 'hull-login__modal':true});
```
Comment (oliger) [inline src/components/overlay/index.jsx]: do we need to use `cx` for this?

File: src/components/sections/log-in-section.jsx [hunk 5]
```diff
@@ -75,12 +75,17 @@ export default React.createClass({
 
     const styles = getStyles();
 
+    let signup_link;
```
Comment (oliger) [inline src/components/sections/log-in-section.jsx]: can you camelcase it please :)

File: src/components/sections/sign-up-section.jsx [hunk 6]
```diff
@@ -76,13 +76,17 @@ export default React.createClass({
     });
 
     const styles = getStyles();
+    let login_link;
```
Comment (oliger) [inline src/components/sections/sign-up-section.jsx]: can you camelcase it please :)

File: src/components/overlay/index.jsx (hunk 4 above)
Comment (unity) [inline src/components/overlay/index.jsx]: I prefer this to manipulating strings ;p WDYT?

File: webpack.config.js (hunk 3 above)
Comment (unity) [inline webpack.config.js]: Sorry. forgot to re-enable. goal was to have a lighter file to download

File: src/components/overlay/index.jsx (hunk 4 above)
Comment (oliger) [inline src/components/overlay/index.jsx]: I do like `cx` but in this case `this.props.className + ' hull-login__modal'` seems clearer. Not a big deal :).

File: npm-debug.log (hunk 2 above)
Comment (unity) [inline npm-debug.log]: :heavy_check_mark:

File: webpack.config.js (hunk 3 above)
Comment (unity) [inline webpack.config.js]: :heavy_check_mark:

File: src/components/sections/log-in-section.jsx (hunk 5 above)
Comment (unity) [inline src/components/sections/log-in-section.jsx]: :heavy_check_mark:

File: src/components/sections/sign-up-section.jsx (hunk 6 above)
Comment (unity) [inline src/components/sections/sign-up-section.jsx]: :heavy_check_mark:

File: src/lib/engine.js (hunk 1 above)
Comment (unity) [inline src/lib/engine.js]: :heavy_check_mark:

File: src/components/ship.jsx [hunk 7]
```diff
@@ -66,25 +69,55 @@ export default React.createClass({
...
+      let b = <a href='#' className='hull-login__button hull-login__button--log-out' onClick={this.props.actions.logOut}>{translate('Log out')}</a>
+      buttons.push(b);
+
+
     } else {
-      l1 = <a href='javascript: void 0;' onClick={this.props.actions.activateLogInSection}>{translate('Log in')}</a>
-      l2 = <a href='javascript: void 0;' onClick={this.props.actions.activateSignUpSection}>{translate('Sign up')}</a>
+
+      if(this.state.shipSettings.show_login){
+        let b= <a href='#' key='log-in' className='hull-login__button hull-login__button--log-in' onClick={this.props.actions.activateLogInSection}>{translate('Log in')}</a>
```
Comment (citelao) [inline src/components/ship.jsx]: `b = <a`

**Comment by oliger** [comment]
@unity Beside the linting errors, it looks good to merge. :+1:

File: manifest.json [hunk 8]
```diff
@@ -2,47 +2,38 @@
   "name": "Login",
   "description": "A complete user registration and social login modal",
   "picture": "picture.png",
+  "index": "ship.js",
```
Comment (oliger) [inline manifest.json]: That means we can remove `src/ship.html`?

File: src/components/ship.jsx [hunk 9]
```diff
@@ -35,6 +35,10 @@ export default React.createClass({
     return `ship-${this.state.ship.id}`;
   },
 
+  getResetStyles() {
+    return this.state.shipSettings && this.state.shipSettings.appearance && this.state.shipSettings.reset_styles;
```
Comment (oliger) [inline src/components/ship.jsx]: `this.state.shipSettings.reset_styles` -> `this.state.shipSettings.appearance.reset_styles` ?

File: manifest.json (hunk 8 above)
Comment (unity) [inline manifest.json]: Yes

**Pull request merged**